wordcounter/wordCounter.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+const chunkSize int32 = 32 * 1024
+
+
+func WordCounter(filename string, words []Word, waitGroup *sync.WaitGroup, out chan<- FileWords) {
+	defer waitGroup.Done()
+
+	fmt.Sprintf("wordReader started for %s", filename)
+
+	// Attempt to open the file
+	file, err := os.Open(filename)
+	defer file.Close()
+
+	if err != nil {
+		fmt.Sprintf("Error opening file %s", filename)
+		return
+	}
+
+	// Start processing data from the file
+	for {
+		buf := make([]byte, chunkSize) // Create a buffer
+		bytes, err := file.Read(buf)   // Read data into the buffer
+		if bytes > 0 {
+			str := string(buf) // Convert bytes to a string
+			wordsFromFile := ConvertStringToArray(str)
+			wordChannel := make(chan Word)
+			wg := sync.WaitGroup{}
+			for _, word := range words {
+				wg.Add(1)
+				go CountCharaterFromWords(word, wordsFromFile, &wg, wordChannel)
+			}
+			go MonitorWordsChannel(&wg, wordChannel)
+			var tempWords []Word
+			for rs := range wordChannel {
+				tempWords = append(tempWords, rs)
+			}
```

Comment:
this will create a deadlock. For normal channel (not buffered), after you write to channel, it will block write, and wait for read to unblock it.

> By default, sends and receives block until the other side is ready. This allows goroutines to synchronize without explicit locks or condition variables.
> https://tour.golang.org/concurrency/2